### PR TITLE
Feature/mtk 310 modify get categories

### PR DESCRIPTION
**Pull request by arnesuscharles**
# Problems

- Complex query gets called every time Get Product List API called, even the result is probably the same every time it gets called

# Purposes

- Add redis mechanism

# Changes

- Create new API to Get Product List, with redis mechanism inside the function 

# Checklist
These are the checklists to make sure your PR is ready to be reviewed.
- [x] Unit testing

# Issue Ticket
Related jira ticket: 
- [MTK-310](https://mitrais-dev.atlassian.net/browse/MTK-310)

### CODE REVIEW

File: domain/product/repository/postgres/postgres_category.go [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+	aliasProductTable := "\"products\""
+	aliasProductCategoryRelationsTable := "\"product_category_relations\""
+	selectCategoryColumn := fmt.Sprintf("%s.id, %s.name, %s.is_active", aliasCategoryTable, aliasCategoryTable, aliasCategoryTable)
+
+	tx := p.db.WithContext(ctx).
+		Select(selectCategoryColumn).
+		Joins(fmt.Sprintf(" INNER JOIN %s on %s.category_id = %s.id ", productCategoryRelationsTable, aliasProductCategoryRelationsTable, aliasCategoryTable)).
+		Joins(fmt.Sprintf(" INNER JOIN %s on %s.product_id = %s.id ", productTable, aliasProductCategoryRelationsTable, aliasProductTable)).
+		Where(fmt.Sprintf(" %s.store_id = ? ", aliasProductTable), input.StoreID).
+		Where(fmt.Sprintf("%s.is_active = ?", aliasCategoryTable), !input.IsIncludeDeactivated).
```
Comment (nurcholisnanda) [inline domain/product/repository/postgres/postgres_category.go]: this IsIncludeDeactivated, does it mean that if we include deactivated we should get all data for all data including both is_active true and false, or should only show is_active false only?

File: domain/product/repository/postgres/postgres_category.go [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+		Group(selectCategoryColumn).
+		Order(fmt.Sprintf("%s.name ASC", aliasCategoryTable))
+
+	if err := tx.Find(&categories).Error; err != nil {
```
Comment (nurcholisnanda) [inline domain/product/repository/postgres/postgres_category.go:35]: I think we agree to still return the error, even if the record is not found, and handle it in service layer, isn't it?

File: domain/product/repository/postgres/postgres_category.go (hunk 1 above)
Comment (arnesuscharles) [reply domain/product/repository/postgres/postgres_category.go]: yes, updated mas, if it is include deactivated, should include all the data

File: domain/product/repository/postgres/postgres_category.go (hunk 2 above)
Comment (arnesuscharles) [reply domain/product/repository/postgres/postgres_category.go:35]: updated also for this one, it should return all error from the repository

**Review by rammdhanmitrais — APPROVED** [review]
gltm

**Review by nurcholisnanda — APPROVED** [review]
LGTM

**Pull request merged**